### DISCUSSION TITLE
dunfell-l4t-r32.5.0 on Jetson Xavier NX 16GB

### DISCUSSION THREAD

**Discussion by chhg82** [OE4T/meta-tegra#1016] [Q&A]
Hi

Is there a way to run the older `dunfell-l4t-r32.5.0` branch on the `jetson-xavier-nx-devkit-emmc` machine with an NX 16GB?
The reason I ask is that the current release of the Mathworks GPU coder is only compatible with Jetpack V4.5.1.

Thanks!

**Comment by dwalkes** [comment]
I'd try backporting https://github.com/OE4T/meta-tegra/commit/5edf038c04a26d1ed1e8ea2894fdfcc04a6a17e2 if you haven't already, I'm not sure what if anything else is needed.  Others may chime in if they know.

**Comment by madisongh** [comment]
I don't think NVIDIA added support for the 16GB NX until R32.7.2.

**Reply by dwalkes** [reply]
Yep, see https://developer.nvidia.com/embedded/linux-tegra-r3272.

**Comment by chhg82** [comment]
I found a patch `Overlay to support Jetson Xavier NX 16GB` for [R32.5.2](https://developer.nvidia.com/embedded/linux-tegra-r3251). There are the files `tegra194-mb1-bct-memcfg-p3668-0001-a00.cfg` and `tegra194-a02-bpmp-p3668-a00.dtb` provided.

**Comment by chhg82** [comment]
The NX 16GB is now running the dunfell-l4t-r32.5.0 branch.
The only thing I did in the extracted tegraflash folder was replacing the file `jetson-xavier-nx-devkit-emmc.cfg` with `tegra194-mb1-bct-memcfg-p3668-0001-a00.cfg` from the patch. Why isn't the `tegra194-a02-bpmp-p3668-a00.dtb` file from the patch needed?
@madisongh: is the dunfell-l4t-r32.5.0 branch dead, or would it still make sense to include this patch (also in dunfell-l4t-r32.6.1)?

**Reply by madisongh** [reply]
Glad to hear it's working OK.  I'm fine with adding this and the other BSP overlays to those branches, and was planning on looking into it when I have some spare time, but if you want to submit a PR with that change, that would be fine, too.

**Reply by chhg82** [reply]
Do you have any idea why it also runs without dtb?

**Reply by madisongh** [reply]
I don't know much about what's in the BPMP device trees... could be the existing settings are close enough to what the BPMP needs to run.

**Reply by chhg82** [reply]
I integrated the patch in `tegra-bootfiles_32.5.2.bbappend`:

```
#SRC_URI not working, tegra-shared-binaries.inc calls do_fetch[noexec] = "1"
CFG_FILE := "tegra194-mb1-bct-memcfg-p3668-0001-a00.cfg"
CFG_FILE_PATH := "${THISDIR}/files/${CFG_FILE}"
DTB_FILE := "tegra194-a02-bpmp-p3668-a00.dtb"
DTB_FILE_PATH := "${THISDIR}/files/${DTB_FILE}"

do_install_append() {
    install -m 0644 ${CFG_FILE_PATH} ${D}${datadir}/tegraflash/${MACHINE}.cfg
    install -m 0644 ${DTB_FILE_PATH} ${D}${datadir}/tegraflash/${DTB_FILE}
}
```
I hope you can use a part of it!?

**Reply by chhg82** [reply]
My next challenge is whether I can run the target in MODE_20W_6CORE. This is not yet available in R32.5.2... :-)
